chcemvediet/apps/inforequests/tests/test_models/test_action.py

```diff
@@ -458,28 +475,79 @@ def test_has_deadline_has_applicant_deadline_and_has_obligee_deadline_methods(se
-                # With explicitly set deadline even if action type does not set any by default
-                (Action.TYPES.ADVANCEMENT,            True,  False,     False, dict(deadline=3)),
-                (Action.TYPES.AFFIRMATION,            True,  False,     False, dict(deadline=3)),
-                (Action.TYPES.REVERSION,              True,  False,     False, dict(deadline=3)),
-                (Action.TYPES.EXPIRATION,             True,  False,     False, dict(deadline=3)),
-                (Action.TYPES.APPEAL_EXPIRATION,      True,  False,     False, dict(deadline=3)),
                 )
         # Make sure we are testing all defined action types
         tested_action_types = set(a for a, _, _, _, _ in tests)
         defined_action_types = Action.TYPES._inverse.keys()
         self.assertItemsEqual(tested_action_types, defined_action_types)
 
-        inforequest = self._create_inforequest()
-        branch = self._create_branch(inforequest=inforequest)
         for action_type, has_deadline, has_applicant_deadline, has_obligee_deadline, extra_args in tests:
-            action = self._create_action(branch=branch, type=action_type, **extra_args)
-            self.assertEqual(action.has_deadline, has_deadline)
+            action = self._create_action(type=action_type, delivered_date=naive_date(u'2010-10-05'), **extra_args)
+            self.assertEqual(bool(action.deadline), has_deadline)
             self.assertEqual(action.has_applicant_deadline, has_applicant_deadline)
             self.assertEqual(action.has_obligee_deadline, has_obligee_deadline)
 
+    def test_has_obligee_deadline_snooze_missed_property(self):
+        with self._test_deadline_missed_aux(type=Action.TYPES.REQUEST, snooze=naive_date(u'2010-10-16')) as action:
+            timewarp.jump(local_datetime_from_local(u'2010-10-16 10:33:00'))
+            self.assertFalse(action.has_obligee_deadline_snooze_missed)
+        with self._test_deadline_missed_aux(type=Action.TYPES.REQUEST, snooze=naive_date(u'2010-10-16')) as action:
+            timewarp.jump(local_datetime_from_local(u'2010-10-17 10:33:00'))
+            self.assertTrue(action.has_obligee_deadline_snooze_missed)
+
+    def test_can_applicant_snooze_property(self):
+        with self._test_deadline_missed_aux(type=Action.TYPES.REQUEST) as action:
+            timewarp.jump(local_datetime_from_local(u'2010-10-13 10:33:00'))
+            self.assertFalse(action.can_applicant_snooze)
+        with self._test_deadline_missed_aux(type=Action.TYPES.REQUEST) as action:
+            timewarp.jump(local_datetime_from_local(u'2010-10-14 10:33:00'))
+            self.assertTrue(action.can_applicant_snooze)
+        with self._test_deadline_missed_aux(type=Action.TYPES.REQUEST) as action:
+            timewarp.jump(local_datetime_from_local(u'2010-10-19 10:33:00'))
+            self.assertFalse(action.can_applicant_snooze)
+
+    def test_get_extended_type_display_property(self):
```

Comment:
Nechceme pridat aj test pre class metodu `create`?

Reply:
:ok:

Reply:
:ok:

Reply:
@viliambalaz Pozor v `_create_action` nepouzivas `Action.create` ale `Action.objects.create`, co je zhruba to iste ako volanie konstruktora `Action(*args, **kwargs)` vo vnutri `Action.create`.

Cize v tomto teste treba otestovat to, co `Action.create` implementuje navyse oproti standardnemu konstruktoru. Cize ano, kontretne ze funguju argumenty `advanced_to` a `attachments`. Narokom je funkcia `Action.create` priamo na modeli a nie na querysete, pretoze na querysete sa neda pouzit `@after_saved(action)` a po savnuti objektu ulozit referencie na neho.

Tak ako si to spravil je OK.